src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java

```diff
@@ -939,6 +939,56 @@ public void testOutOfRangesArguments(final String[] argv) throws Exception {
         return result;
     }
 
+    @CommandLineProgramProperties(
+            summary = "tool with bounded collection argument",
+            oneLineSummary = "bounded collection argumen",
+            programGroup = TestProgramGroup.class
+    )
+    public class BoundedCollection {
+        @Argument(doc = "Integer collection in the range [0, 30]",
+                optional = true,
+                minValue = 0, maxValue = 30,
+                minRecommendedValue = 10, maxRecommendedValue = 15)
+        public List<Integer> intListArg = new ArrayList<>();
+    }
+
+    @DataProvider(name="goodBoundedCollectionArgs")
+    public Object[][] goodBoundedCollection() {
+        return new Object[][]{
+                {new String[]{"--intListArg", "1"}, new Integer[]{ 1 } },
+                {new String[]{"--intListArg", "19", "--intListArg", "2"}, new Integer[]{ 19, 2 } },
+                {new String[]{"--intListArg", "0", "--intListArg", "18", "--intListArg", "27" }, new Integer[]{ 0, 18, 27 } },
+        };
+    }
+
+    @Test(dataProvider = "goodBoundedCollectionArgs")
+    public void testGoodBoundedCollections(final String[] argv, final Integer[] expectedIntegers) throws Exception {
```

Comment:
Only because Integer is the type of the value in the argument, so it makes the Assert overload unambiguous. I did remove the unnecessary .getInt calls from the Asserts though.